### PR TITLE
CSC Output, Idemp Err Caching, Cred/Info Format

### PR DESCRIPTION
**Pull request by akutz**
This patch includes the following changes:

1. csc output involving volume info and plug-in info now emit the attribute and manifest maps on the same line as the other data
2. **Error caching has been removed from the idempotency interceptor.** ~~The idempotency interceptor now only caches errors and bypasses idempotency on subsequent RPC invocations if the feature is enabled with `X_CSI_IDEMP_CACHE_ERRORS=true`.~~
3. The formats of `X_CSI_USER_CREDENTIALS`, `X_CSI_PLUGIN_INFO`, and `X_CSI_SUPPORTED_VERSIONS` have all changed from space-delimited to comma-delimited.

### CODE REVIEW

File: csc/cmd/formats.go [hunk 1]
```diff
@@ -21,5 +21,6 @@ const supportedVersionsFormat = `{{range $v := .SupportedVersions}}` +
 
 // pluginInfoFormat is the default Go template for emitting a
 // csi.GetPluginInfoResponse
-const pluginInfoFormat = `{{printf "%q\t%q\n" .Name .VendorVersion}}` +
-	`{{range $k, $v := .Manifest}}{{printf "%q=%q\n" $k $v}}{{end}}`
+const pluginInfoFormat = `{{printf "%q\t%q" .Name .VendorVersion}}` +
+	`{{range $k, $v := .Manifest}}{{printf "\t%q=%q" $k $v}}{{end}}` +
+	`{{"\n"}}`
```
Comment (codenrhoden) [inline csc/cmd/formats.go]: Why the change to have this all on one line? Just for potential to parse it easily in a script or other program? With a long manifest, this could get large. I liked the tabular format.

File: interceptors_idempotency.go [hunk 2]
```diff
@@ -175,18 +185,20 @@ func (i *idempotencyInterceptor) controllerPublishVolume(
...
-	}()
-
-	// If the method has been marked in error then it means a previous
-	// call to this function returned an error. In these cases a
-	// subsequent call should bypass idempotency.
-	if _, ok := lock.methodInErr[info.FullMethod]; ok {
-		return handler(ctx, req)
+	if i.opts.cacheErrs {
+		// At the end of this function check for a response error. If there
+		// is an error that isn't VolumeNotFound then mark this method as
```
Comment (codenrhoden) [inline interceptors_idempotency.go]: I should know this, but why the stated exception for `VolumeNotFound`? If a controllerPublishVolume call returns VolumeNotFound, it seems like if the call is retried, we would want to bypass idempotency and defer to the SP.

File: csc/cmd/formats.go (hunk 1 above)
Comment (akutz) [reply csc/cmd/formats.go]: Hi @codenrhoden,

It's still tabular, it's just now that there is a single entry **per** line. If there are more than two fields one can assume all remaining fields belong to the manifest. I agree it can get long, but I also think it won't be that much more ugly than on its own line if the manifest is really that large.

File: interceptors_idempotency.go (hunk 2 above)
Comment (akutz) [reply interceptors_idempotency.go]: Hi @codenrhoden,

Primarily to be consistent with the rest of it, but also because the debt of **not** bypassing idempotency for this is rather small since it will work by determining the volume does not exist.
